### ISSUE TITLE
Getting index property of option element fails if no containing select element

### ISSUE THREAD

**Issue by TimBarham** [tmpvar/jsdom#1403]
If you have an `option` element that has no containing `select` element (for example, you just created it using `document.createElement()`), trying to get its `index` property will throw an exception on [this](https://github.com/tmpvar/jsdom/blob/master/lib/jsdom/living/nodes/HTMLOptionElement-impl.js#L81) line:

```
    return Array.prototype.indexOf.call(closest(this, "select").options, idlUtils.wrapperForImpl(this));
                                                               ^
TypeError: Cannot read property 'options' of null
    at HTMLOptionElementImpl.index (D:\WebStormProjects\DomNodes\node_modules\jsdom\lib\jsdom\living\nodes\HTMLOptionElement-impl.js:81:64)
```

In this same scenario, all browsers I tried (Chrome, Firefox, IE, Edge) return `0`.

**Comment by domenic** [comment]
Can you provide a minimal code example triggering this?

**Comment by TimBarham** [comment]
This will repro it:

``` js
var jsdom = require('jsdom');

jsdom.env('', function (err, win) {
    var optionElement = win.document.createElement('option');
    console.log(optionElement.index);
});
```

And I should mention I'm using latest jsdom source from github.

**Comment by domenic** [comment]
Hmm. Can you test with the published version from npm? We don't generally support installation from GitHub.

**Comment by TimBarham** [comment]
Oh, actually, that was my bad - this project actually is using the latest published version from npm (8.0.4) (I was also concurrently working on something else in a git clone jsdom, and forgot that the project where I was seeing the error wasn't actually using that).

**Comment by domenic** [comment]
OK cool. Will try to fix tonight-ish...

**Comment by TimBarham** [comment]
Thanks @domenic.

BTW, I'm just encountered similar errors with other properties:

* Getting `cellIndex` of `td` that isn't contained in a `tr`.
* Getting `headers` of a `td` that doesn't have a parent.
* Getting 'sectionRowIndex' of a `tr` that doesn't have a parent.

(BTW... I'm hitting nearly every property of every element type and a bunch of other types with this code, and these are the only errors I have found).

Do you want separate issues for those? In all cases, the property and tag name can be substituted into my code snippet above to repro.

**Comment by domenic** [comment]
One issue is fine! Great finds, and glad to hear that this is all that exhaustive tests have managed to uncover :)

**Comment by domenic** [comment]
Fixed in 8.1.0!

**Comment by TimBarham** [comment]
Fantastic, thanks @domenic!
